components/raftstore-v2/src/operation/query/mod.rs

```diff
@@ -23,13 +27,31 @@ use crate::{
     router::{QueryResChannel, QueryResult},
 };
 
+mod follower_read_index;
 mod local;
+mod read_index;
+mod replica_read;
 
-impl<'a, EK: KvEngine, ER: RaftEngine, T> PeerFsmDelegate<'a, EK, ER, T> {
+impl<'a, EK: KvEngine, ER: RaftEngine, T: raftstore::store::Transport>
+    PeerFsmDelegate<'a, EK, ER, T>
+{
     #[inline]
     pub fn on_query(&mut self, req: RaftCmdRequest, ch: QueryResChannel) {
         if !req.has_status_request() {
-            unimplemented!();
+            let mut resp = RaftCmdResponse::default();
+            let term = self.fsm.peer().term();
+            cmd_resp::bind_term(&mut resp, term);
+            let policy = self.fsm.peer_mut().inspect(&req);
```

Comment:
I don't think `inspect` is necessary anymore. Because we are expecting this code path is for read only. So a lot of checks are unnecessary or can return early.

Reply:
But still we need to differentiate ReadStale, Get, ReadIndex.  To me it's better to create a new inspect and copy some of the code from it.